### PR TITLE
DOC: Highlight pre-commit section

### LINKED ISSUE #37358: DOC mention pre-commit as a way to run (some) code checks
xref this comment https://github.com/pandas-dev/pandas/pull/37240#discussion_r510518893

> as a followup we might want to update the contributing docs to put a pointer to these types of checks here (e.g. rather than just mentioning code_checks.sh) esp as we are using these a lot more now

### PR DESCRIPTION
**Pull request by hongshaoyang**
- [x] closes #37358 
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by hongshaoyang** [comment]
I noticed that https://dev.pandas.io/ is not up to date, is this intended?

**Comment by hongshaoyang** [comment]
~~I noticed that https://dev.pandas.io/ is not up to date, is this intended?~~

it seems like this is intentional and that latest docs from `master` are at https://pandas.pydata.org/docs/dev/, see #37248.

**Comment by jreback** [comment]
looks fine to me. @MarcoGorelli if you can have a look see

File: doc/source/development/contributing.rst [hunk 1]
```diff
@@ -712,7 +751,7 @@ Python (PEP8 / black)
 pandas follows the `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_ standard
 and uses `Black <https://black.readthedocs.io/en/stable/>`_ and
 `Flake8 <http://flake8.pycqa.org/en/latest/>`_ to ensure a consistent code
-format throughout the project.
+format throughout the project. We encourage you to use :ref:`pre-commits <contributing.pre-commit>`.
```
Comment (MarcoGorelli) [inline doc/source/development/contributing.rst:754]:
```suggestion
format throughout the project. We encourage you to use :ref:`pre-commit <contributing.pre-commit>`.
```

File: doc/source/development/code_style.rst [hunk 2]
```diff
@@ -12,8 +12,9 @@ pandas code style guide
 pandas follows the `PEP8 <https://www.python.org/dev/peps/pep-0008/>`_
 standard and uses `Black <https://black.readthedocs.io/en/stable/>`_
 and `Flake8 <https://flake8.pycqa.org/en/latest/>`_ to ensure a
-consistent code format throughout the project. For details see the
-:ref:`contributing guide to pandas<contributing.code-formatting>`.
+consistent code format throughout the project. We encourage you to use
+:ref:`pre-commits <contributing.pre-commit>` to automatically run ``black``,
```
Comment (MarcoGorelli) [inline doc/source/development/code_style.rst:16]:
```suggestion
:ref:`pre-commit <contributing.pre-commit>` to automatically run ``black``,
```

File: doc/source/development/contributing.rst [hunk 3]
```diff
@@ -638,7 +638,46 @@ In addition to ``./ci/code_checks.sh``, some extra checks are run by
...
+
+    pip install pre-commit
+
+and then running::
+
+    pre-commit install
+
+from the root of the pandas repository. Now all of the styling checks will be
+run each time you commit changes without your needing to run each one manually.
+In addition, using this pre-commit hook will also allow you to more easily
```
Comment (MarcoGorelli) [inline doc/source/development/contributing.rst:661]:
```suggestion
In addition, using ``pre-commit`` will also allow you to more easily
```

File: doc/source/development/contributing.rst [hunk 4]
```diff
@@ -638,7 +638,46 @@ In addition to ``./ci/code_checks.sh``, some extra checks are run by
...
+remain up-to-date with our code checks as they change.
+
+Note that if needed, you can skip these checks with ``git commit --no-verify``.
+
+If you don't want to use ``pre-commit`` as part of your workflow, you can still use it
+to run its checks by running::
```
Comment (MarcoGorelli) [inline doc/source/development/contributing.rst:667]:
```suggestion
If you don't want to use ``pre-commit`` as part of your workflow, you can still use it
to run its checks with::
```

File: doc/source/development/contributing.rst [hunk 5]
```diff
@@ -638,7 +638,46 @@ In addition to ``./ci/code_checks.sh``, some extra checks are run by
...
+
+    pre-commit run --files <files you have modified>
+
+without having to have done ``pre-commit install`` beforehand.
```
Comment (MarcoGorelli) [inline doc/source/development/contributing.rst:671]:
```suggestion
without needing to have done ``pre-commit install`` beforehand.
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Looks good, thanks, just suggesting some minor changes

**Comment by hongshaoyang** [comment]
Thanks @MarcoGorelli , applied your suggestions.

**Comment by jreback** [comment]
lgtm. cc @MarcoGorelli merge when ready

**Comment by MarcoGorelli** [comment]
Thanks @hongshaoyang

**Pull request merged**